TreeView: Update Selection & Customization Demo Desc. (#32664)
DevExpress/DevExtreme: apps/demos/Demos/TreeView/ItemSelectionAndCustomization/React/description.md

> DevExtreme [TreeView](/Documentation/Guide/UI_Components/TreeView/Overview/) ships with configurable selection capabilities. Users can click nodes or item checkboxes to select. This demo allows you to modify selection behavior using controls below the TreeView. A DevExtreme [List](/Documentation/Guide/UI_Components/List/Overview/) to the right of the TreeView displays selected items in a flat (non-hierarchical) list.

Reviewer: Copilot has a point. But I'll try to optimize the intro instead. What do you think?

```suggestion
The DevExtreme [TreeView](/Documentation/Guide/UI_Components/TreeView/Overview/) allows users to select nodes (on click or using checkboxes). In this demo, selected items appear as a flat (non-hierarchical) [List](/Documentation/Guide/UI_Components/List/Overview/).
```

<!--split-->

> This demo allows you to specify a [selection mode](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#selectionMode) for enabled and [disabled nodes](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#disabledNodeSelectionMode). You can configure the visibility of [selection checkboxes](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#showCheckBoxesMode), including the Select All checkbox. You can also toggle [selection on click](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#selectByClick) and [recursive selection](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#selectNodesRecursive).

Reviewer: We do not use articles with "mode". Also, I suggest to get rid of repetative intros. To do that, I am moving one of the sentences from the previous parapraph:

```suggestion
Use controls below the TreeView to modify selection behavior as follows:

- Specify [selection mode](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#selectionMode) for enabled and [disabled nodes](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#disabledNodeSelectionMode)
- Toggle visibility of [selection checkboxes](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#showCheckBoxesMode) (including the Select All option)
- Disable [selection on click](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#selectByClick) 
- Enable [recursive selection](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#selectNodesRecursive).
```

Please check if my list options contrsdict the default demo configuration. You can also try to come up with better synonyms for the two latter options.

Author: The last two should be flipped. Perhaps we can say "Allow selection on click"?

```suggestion
Use controls below the TreeView to modify selection behavior as follows:

- Specify [selection mode](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#selectionMode) for enabled and [disabled nodes](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#disabledNodeSelectionMode)
- Toggle visibility of [selection checkboxes](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#showCheckBoxesMode) (including the Select All option)
- Allow [selection on click](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#selectByClick) 
- Disable [recursive selection](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#selectNodesRecursive).
```

> To synchronize the List with TreeView, this demo implements a [selection change](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#onSelectionChanged) handler. The handler calls [getSelectedNodes()](/Documentation/ApiReference/UI_Components/dxTreeView/Methods/#getSelectedNodes) to update the List data source.

Reviewer: I am not completely sure we need to describe such specifics here... Can we just remove this paragraph? But if we do, we need to get rid of repetitions: "To synchronize the List with TreeView" - "to update the List data source" - this is the same idea from where I am standing.

Author: Yeah, we probably can skip describing List updating specifics

> TreeView allows you to update node selection states programmatically. You can select/unselect items using [node keys](/Documentation/ApiReference/UI_Components/dxTreeView/Methods/#selectItemkey), [data objects](/Documentation/ApiReference/UI_Components/dxTreeView/Methods/#selectItemitemData), or [DOM elements](/Documentation/ApiReference/UI_Components/dxTreeView/Methods/#selectItemitemElement). To initialize nodes in the selected state, configure [selection fields](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#selectedExpr) in item objects (default: **selected**).

Reviewer: ```suggestion
The TreeView allows you to update node selection states programmatically. You can select/deselect items by [node keys](/Documentation/ApiReference/UI_Components/dxTreeView/Methods/#selectItemkey), [data objects](/Documentation/ApiReference/UI_Components/dxTreeView/Methods/#selectItemitemData), or [DOM elements](/Documentation/ApiReference/UI_Components/dxTreeView/Methods/#selectItemitemElement). To initially select nodes, configure [selection fields](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#selectedExpr) within item objects (default: **selected**).
```

> You can use selection and other data fields to customize items as needed. This demo implements [itemRender](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#itemRender) to combine text from multiple fields and does not specify a [text](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/items/#text) field in the TreeView data source.

Reviewer: ```suggestion
You can use data fields to customize items as your requirements dictate. This demo implements [itemRender](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/#itemRender) to merge text from multiple field: you do not need to specify the [text](/Documentation/ApiReference/UI_Components/dxTreeView/Configuration/items/#text) field in the TreeView data source.
```
